runtime/lib/formatter/PropelObjectFormatter.php

```diff
@@ -39,16 +39,21 @@ public function format(PDOStatement $stmt)
             $pks = array();
             $objectsByPks = array();
             while ($row = $stmt->fetch(PDO::FETCH_NUM)) {
-                $object = $this->getAllObjectsFromRow($row);
-                $pk = $object->getPrimaryKey();
 
                 if (false === Propel::isInstancePoolingEnabled()) {
+
+                    list($object, $col) = call_user_func(array($this->peer, 'populateObject'), $row);
+                    $pk = $object->getPrimaryKey();
+
                     if (isset($objectsByPks[$pk])) {
                         $this->mainObject = $objectsByPks[$pk];
-                        $object = $this->getAllObjectsFromRow($row);
                     }
+                    $object = $this->getAllObjectsFromRow($row);
```

Comment:
let me sum up your PR:

the actual issue is, that this `$object` does not return the `$pk` you expected on `getPrimaryKey()` call, when instance-pooling is disabled?